[9.0][FIX] mandatory field not displayed

elouan-lb opened the pull request:
Change the `api.onchange`, which is only suggestive, for an SQL constraint more restrictive.

Will do the same for the `info_session` field in `beesdoo_easy_my_coop` of Obeesdoo repo when PR merged.

houssine78 commented on easy_my_coop/models/company.py:
```diff
@@ -81,17 +81,12 @@ def _get_base_logo(self):
-    @api.onchange('financial_risk_approval_required')
-    def onchange_financial_risk_approval_required(self):
-        if self.financial_risk_approval_required:
-            self.display_financial_risk_rules_approval = True
```
@elouan-lb you can leave @api.onchange method as it act more like helpers than constraint. It auto check the checkbox when the related checkbox is checked. So we should better keep those methods aside the constraints.

robinkeunen replied:
👍

elouan-lb replied:
I reverted the changes, thanks for the review.

robinkeunen commented on easy_my_coop/models/company.py:
```diff
@@ -81,17 +81,12 @@ def _get_base_logo(self):
+    AND ((financial_risk_approval_required=FALSE AND display_financial_risk_approval=FALSE) \
+    OR display_financial_risk_approval=TRUE))',
+     "Approval can't be mandatory and not displayed.")
+    ]
```
Il y a petit problème de formatting de ton IDE ici. Ça rend la lecture de la requête difficile. Tu peux utiliser des triple-quotes pour écrire des requêtes de plusieurs lignes.

```python
long_text = """
This is a long
multiline
string
"""
```

elouan-lb replied:
Voilà qui est fait, j'ai utilisé une indentation qui ne semble pas standard mais me paraît plus lisible, qu'en penses-tu ?

houssine78 replied:
la requête est lancée à chaque mise à jour du module ?

robinkeunen commented on easy_my_coop/migrations/9.1.0.0.2/pre-migration.py:
```diff
@@ -0,0 +1,15 @@
+    if not version:
+        return
+
+    cr.execute('UPDATE res_company \
```
Quand sera exécuté ce code? Faudra faire tourner un _openupgrade_ ou un update de la base suffit?

elouan-lb replied:
La mise à jour du module suffit à executer la commande SQL !

houssine78 replied:
la requête est lancée à chaque mise à jour du module ?

elouan-lb replied:
Non, simplement à la suivante (grâce au numéro de version), si ne me trompe pas (@remytms c'est bien ça ?)

houssine78 replied:
ok pour moi dans ce cas !

elouan-lb commented:
PR merged, forgot to update the branch before merging locally.

robinkeunen commented:
> PR merged, forgot to update the branch before merging locally.

Not sure of what you mean by that.

elouan-lb commented:
> > PR merged, forgot to update the branch before merging locally.
> 
> Not sure of what you mean by that.

I rebased and squashed the commits, then merged them on 9.0 and pushed it, but I forgot to push the squashed history on this branch before.

houssine78 commented:
@elouan-lb it seems you closed the PR but you didn't merge it. Do I'm missing something ?

elouan-lb commented:
As written previously, I did merge it (see f406155d5c7250b773a70ec7b8b4d6dd6bdd3da7) !
